hwy/ops/arm_neon-inl.h

```diff
@@ -6484,6 +6484,320 @@ HWY_API VFromD<D> SlideDownLanes(D d, VFromD<D> v, size_t amt) {
   return detail::SlideDownLanes(v, amt);
 }
 
+// ------------------------------- WidenHighMulAdd
+
+#ifdef HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#undef HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#else
+#define HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#endif
+
+template<class D, HWY_IF_U32_D(D)>
+HWY_API Vec128<uint64_t> WidenHighMulAdd(D /* tag */, Vec128<uint32_t> mul,
```

Comment:
By convention, D describes the return type. Would you mind changing all the functions? Here toIF_U64. And to make this clear + code more generic, we can write the return type as VFromD<D>.

Another important bit is to support partial vectors. Hard-coding Vec128<uint32_t> means we can only handle full vectors. All that's required is to make the arguments depend on D. For example, we can set `VN = VFromD<RepartitionToNarrow<D>>` and use that as the arg type.

Reply:
Done. By the way, it is not clear enough to me how does partial vectors in highway.

I changed here to MulAdd, because it fails tests for sometimes for upper half, sometimes for lower. I didn't get where it is has to be where vec length = 1.
```c++
template<class D, HWY_IF_F16(TFromD<D>), HWY_IF_LANES_D(D, 2),
    class DW = RepartitionToWide<D>>
HWY_API VFromD<DW> WidenHighMulAdd(D /* tag */, VFromD<D> mul,
                                   VFromD<D> x, VFromD<DW> add) {
  const DW dw;
  auto ho = Vec64<float32_t>(vfmlal_high_f16(add.raw, mul.raw, x.raw));
  return UpperHalf(dw, ho);
//  return MulAdd(add, PromoteUpperTo(dw, mul), PromoteUpperTo(dw, x));
}
```

Reply:
Is your question what to do if Lanes()==1? I believe this does not happen because ForShrinkableVectors's kPow2=1 should be ensuring the minimum lane count is 2.

Reply:
It happens when returning vector `VFromD<DW>` is 1

Reply:
It was quite surprising when tests were failed for UpperHalf and LowerHalf :)